gdpr/tests/test_gdpr_views.py

```diff
@@ -0,0 +1,192 @@
+    user_factory,  # users
+    area_search_factory,  # plotsearch
+    area_search_attachment_factory,  # plotsearch
+):
+    settings.GDPR_API_QUERY_SCOPE = "gdprquery"
+    settings.OIDC_API_TOKEN_AUTH = {
+        "API_AUTHORIZATION_FIELD": "authorization.permissions.scopes",
+    }
+
+    user = user_factory(first_name="Jari", last_name="Kuningas")
+    areasearch = area_search_factory(user=user, description_area="Test")
+    areasearch_attachment = area_search_attachment_factory(
+        user=user, area_search=areasearch, name="king_of_finland.txt"
+    )
+    apirequest_factory = APIRequestFactory()
+    request = apirequest_factory.get(f"/v1/pub/gdpr-api/v1/profiles/{user.uuid}")
+    user_authorization = UserAuthorization(
+        user=user,
+        api_token_payload={
+            "amr": ["suomi_fi"],
+            "authorization": {"permissions": {"scopes": ["gdprquery"]}},
+        },
+    )
+    force_authenticate(request, user=user, token=user_authorization)
+
+    response = MvjGDPRAPIView.as_view()(request, uuid=user.uuid)
+
+    assert response.status_code == status.HTTP_200_OK
+
+    user_data_dicts = {
+        x.get("key"): x.get("value")
+        for x in response.data.get("children")
+        if isinstance(x, dict)
+    }
+    assert user_data_dicts.get("FIRST_NAME") == "Jari"
+    assert user_data_dicts.get("LAST_NAME") == "Kuningas"
+
+    areasearchattachments = _find_matching_dicts(
+        response.data.get("children"), "key", "AREASEARCHATTACHMENT"
+    )
+    areasearchattachment_name = _find_matching_dicts(
+        areasearchattachments, "key", "NAME"
+    )[0]["value"]
+    assert areasearchattachment_name == areasearch_attachment.name
+
+
+@pytest.mark.django_db
+def test_api_get_user_data_invalid_scope(
+    settings,
+    user_factory,  # users
+):
+    settings.GDPR_API_QUERY_SCOPE = "gdprquery"
+    settings.OIDC_API_TOKEN_AUTH = {
+        "API_AUTHORIZATION_FIELD": "authorization.permissions.scopes",
+    }
+
+    user = user_factory()
+    apirequest_factory = APIRequestFactory()
+    request = apirequest_factory.get(f"/v1/pub/gdpr-api/v1/profiles/{user.uuid}")
+    scopes = ["invalidscope"]
+    user_authorization = UserAuthorization(
+        user=user,
+        api_token_payload={
+            "amr": ["suomi_fi"],
+            "authorization": {"permissions": {"scopes": scopes}},
+        },
+    )
+    force_authenticate(request, user=user, token=user_authorization)
+
+    response = MvjGDPRAPIView.as_view()(request, uuid=user.uuid)
+
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+@pytest.mark.django_db
+def test_api_delete_user_data(
+    settings,
+    user_factory,  # users
+    area_search_factory,  # plotsearch
+):
+    settings.GDPR_API_DELETE_SCOPE = "gdprdelete"
+    settings.OIDC_API_TOKEN_AUTH = {
+        "API_AUTHORIZATION_FIELD": "authorization.permissions.scopes",
+    }
+
+    user = user_factory(first_name="Kuningas", last_name="Litmanen")
+    user_id = user.id
```

Comment:
Is this definition necessary? It's only used on line 137 where it could be just `user.id`